amp/WEB-INF/src/org/digijava/module/aim/util/AuditLoggerUtil.java

```diff
@@ -258,7 +258,63 @@ public static void logActivityUpdate(HttpServletRequest request, AmpActivityVers
             throw new RuntimeException(ex);
         }
     }
+    public static Collection<AmpAuditLogger> getFilteredUser(boolean withLogin, Long userid) {
+        try {
+            String qryStr = null;
+            if (!withLogin){
+                qryStr = "select f from " + AmpAuditLogger.class.getName() + " f where action<>'"
+                        + Constants.LOGIN_ACTION + "' and userid=:userid order by modifyDate desc";
+            } else {
+                qryStr = "select f from " + AmpAuditLogger.class.getName() + " f where userid=:userid order by modifyDate desc";
+            }
+            return PersistenceManager.getSession().createQuery(qryStr).setParameter("userid", userid).list();
+        } catch (Exception ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+    public static Collection<AmpAuditLogger> getFilteredTeam(boolean withLogin, String filteredTeam) {
+        try {
+            String qryStr = null;
+            if (!withLogin){
+                qryStr = "select f from " + AmpAuditLogger.class.getName() + " f where action<>'"
+                        + Constants.LOGIN_ACTION + "' and teamname=:filteredTeam order by modifyDate desc";
+            } else {
+                qryStr = "select f from " + AmpAuditLogger.class.getName() + " f where teamname=:filteredTeam order by modifyDate desc";
+            }
+            return PersistenceManager.getSession().createQuery(qryStr).setParameter("filteredTeam", filteredTeam).list();
+        } catch (Exception ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+    
+    public static Collection<AmpAuditLogger> getFilterUserByDate(boolean withLogin,Long userid, Date dateFrom, Date dateTo) {
+        try {
+            String qryStr = null;
+            if (!withLogin){
+                qryStr = "select f from " + AmpAuditLogger.class.getName() + " f where action<>'"
+                        + Constants.LOGIN_ACTION + "' and userid=:userid and modifyDate BETWEEN :dateFrom AND :dateTo order by modifyDate desc";
+            } else {
+                qryStr = "select f from " + AmpAuditLogger.class.getName() + " f where modifyDate BETWEEN :dateFrom AND :dateTo order by modifyDate desc";
+            }
+            Query qry = PersistenceManager.getSession().createQuery(qryStr);
+            qry.setParameter("userid", userid);
+            qry.setParameter("dateFrom", dateFrom);
```

Comment:
i would use date < and > so we can filter dates greater than, smaller than, and between testing if the value is null we dont add it

Reply:
why prefer < and > than between?

Reply:
so you can decide based on the dates selected if you can filtervalues greater than 1.1.2019 or lower than 1.10.2019 or between those dates,